core/modulesource.go

```diff
@@ -421,12 +421,16 @@ func (src *ModuleSource) ResolveContextPathFromModule(ctx context.Context) (cont
 	if !contextFound {
 		// default to restricting to the source root dir, make it abs though for consistency
 		contextAbsPath = moduleRootAbsPath
-	} else {
-		// If context is found, we can create the module root path by joining the 
-		// context path with the module root subpath
-		moduleRootAbsPath = filepath.Join(contextAbsPath, src.AsLocalSource.Value.RootSubpath)
 	}
 
+	sourceSubPath, err := src.SourceSubpathWithDefault(ctx)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to get source subpath: %w", err)
+	}
+	// If context is found, we can create the module root path by joining the
+	// context path with the source root subpath
+	moduleRootAbsPath = filepath.Join(contextAbsPath, sourceSubPath)
```

Comment:
This isn't the `moduleRootAbsPath` anymore - the naming is wrong, and the doc string for this return value is also not entirely right.